### DISCUSSION TITLE
AR Moving camera

### DISCUSSION THREAD

**Discussion by bknill** [pmndrs/xr#186] [Q&A] [answered]
Hey I'm trying to get a similar effect to the old version in the new version.

i.e. With ARCanvas and the example on the site, https://codesandbox.io/s/8w8hm?file=/src/App.tsx

When I move the phone towards the object it tracks in 3D space and I can move closer to the object. 

With the newer `<XR>` component, the camera doesn't appear to track and the object is fixed in position regardless of how the phone moves.  i.e. I want to look closer by moving the camera towards the object etc.  

What are the settings that were in ARCanvas that allowed the camera to move?

Is this depth sensing? Is this something that been removed since ARCanvas?

**Comment by saitonakamura** [comment]
I wonder if it's an issue with `three` version, can you tell me what versions of fiber and three you're using?

**Comment by bknill** [comment]
Latest   "three": "0.144.0",

"@react-three/fiber": "8.7.1", I'll upgrade that see if it helps. 

Is there the source of the original ARCanvas anywhere?

**Reply by bknill** [reply]
I upgraded everything and tried again. Same result.

**Reply by CodyJasonBennett** [reply]
You can see ARCanvas in https://github.com/pmndrs/react-xr/blob/v4/src/XR.tsx#L370, but it's the same as:

```jsx
<ARButton />
<Canvas>
  <XR>
    // ...
  </XR>
</Canvas>
```

I believe the behavior you're describing is a regression since 3.5.x, specifically https://github.com/pmndrs/react-xr/pull/142.

**Reply by bknill** [reply]
Hey @CodyJasonBennett any update on this? Does it need an issue?

**Reply by CodyJasonBennett** [reply]
We published a patch that should help here. Please do create an issue if there are any withstanding issues.

**Comment by bknill** [comment]
I've updated the example with the latest code and the way I've configured it. 

https://codesandbox.io/s/react-xr-simple-ar-demo-forked-kdj1ui

This is doing the same thing, object moves away from camera when you move the camera towards it.

**Reply by saitonakamura** [reply]
Thanks I'll look into it

**Comment by bbohlender** [comment, marked as answer]
@bknill 
Just doing some housekeeping here. This should be fixed in react-three/xr v6. If the issue persists, please create an issue :)
